### PR TITLE
DB: Deprecate - is_built_with_elementor (ED-1765)

### PR DESCRIPTION
**Pull request by Nevoss**
~Depends on #12879~

### CODE REVIEW

File: includes/db.php [hunk 1]
```diff
@@ -344,14 +344,30 @@ public function copy_elementor_meta( $from_post_id, $to_post_id ) {
...
 	 * @access public
 	 *
 	 * @param int $post_id Post ID.
 	 *
 	 * @return bool Whether the post was built with Elementor.
 	 */
 	public function is_built_with_elementor( $post_id ) {
-		return ! ! get_post_meta( $post_id, '_elementor_edit_mode', true );
+		Plugin::$instance->modules_manager
+			->get_modules('dev-tools')
```
Comment (rodolphebertozzo) [inline includes/db.php:356]: ->get_modules( 'dev-tools' )

File: tests/phpunit/elementor/core/admin/test-admin.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	public function test_add_elementor_post_state() {
+		// Arrange
+		$admin = new Admin();
+		$post = $this->factory()->post->create_and_get();
+
+		// Act
+		$result = $admin->add_elementor_post_state( [], $post );
+
+		// Assert
+		$this->assertArrayNotHasKey( 'elementor', $result );
```
Comment (matipojo) [inline tests/phpunit/elementor/core/admin/test-admin.php:44]: Test also the case it should have the state.

File: tests/phpunit/elementor/core/test-documents-manager.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Test_Documents_Manager extends Elementor_Test_Base {
+	public function test_ajax_get_document_config__set_id_build_with_elementor() {
+		// Arrange
+		Plugin::$instance->editor->set_edit_mode( true );
+
+		wp_set_current_user( $this->factory()->create_and_get_administrator_user()->ID );
```
Comment (matipojo) [inline tests/phpunit/elementor/core/test-documents-manager.php:17]: Use something like `act_as_admin`

File: tests/phpunit/elementor/modules/gutenberg/test-module.php [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+use Elementor\Core\Base\Document;
+use Elementor\Testing\Elementor_Test_Base;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Test_Module extends Elementor_Test_Base {
+	public function test_register_elementor_rest_field(  ) {
+		wp_set_current_user($this->factory()->create_and_get_administrator_user()->ID);
```
Comment (matipojo) [inline tests/phpunit/elementor/modules/gutenberg/test-module.php:13]: Use something like `act_as_admin`

File: tests/phpunit/elementor/modules/gutenberg/test-module.php [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+		// Run reset api.
+		$request = new \WP_REST_Request( 'POST', "/wp/v2/posts/{$post->ID}");
+
+		$request->set_body_params([
+			'gutenberg_elementor_mode' => false,
+		]);
+
+		rest_do_request( $request );
+
+		$this->assertEmpty(get_post_meta($post->ID, Document::BUILT_WITH_ELEMENTOR_META_KEY, true));
```
Comment (matipojo) [inline tests/phpunit/elementor/modules/gutenberg/test-module.php:33]: PHPCS

File: tests/phpunit/elementor/modules/history/test-revisions-manager.php [hunk 6]
```diff
@@ -147,12 +148,13 @@ public function test_should_restore_revision() {
...
 		$autosave_post_id = $res['child_id'];
 
-		update_metadata( 'post', $autosave_post_id, '_elementor_edit_mode', true );
+		update_metadata( 'post', $autosave_post_id, Document::BUILT_WITH_ELEMENTOR_META_KEY, 'builder' );
 		update_metadata( 'post', $autosave_post_id, '_elementor_meta_data', 'content' );
 
 		Revisions_Manager::restore_revision( $post_id, $autosave_post_id );
 
 		$this->assertTrue( $this->check_revisions( $post_id, $autosave_post_id ) );
+		$this->assertEquals('builder', get_post_meta($post_id, Document::BUILT_WITH_ELEMENTOR_META_KEY, true));
```
Comment (matipojo) [inline tests/phpunit/elementor/modules/history/test-revisions-manager.php:157]: PHPCS

File: assets/dev/scss/frontend/_compatibility.scss [hunk 7]
```diff
@@ -23,46 +23,3 @@
 #wpadminbar * {
 	font-style: normal;
 }
-
```
Comment (matipojo) [inline assets/dev/scss/frontend/_compatibility.scss:26]: Remove from PR

**Review by matipojo — CHANGES_REQUESTED** [review]

File: tests/phpunit/elementor/core/admin/test-admin.php [hunk 8]
```diff
@@ -10,6 +11,27 @@
...
+
+		$this->pagenow = $pagenow;
+	}
+
+	public function tearDown() {
+		global $pagenow;
+
+		parent::tearDown();
+
+		$pagenow = $this->pagenow;
```
Comment (matipojo) [inline tests/phpunit/elementor/core/admin/test-admin.php:32]: It's needed? the original `tearDown` is supposed to restore all globals.

File: tests/phpunit/elementor/core/admin/test-admin.php [hunk 9]
```diff
@@ -27,12 +49,74 @@ public function test_save_post() {
 	public function test_save_post__when_saved_outside_of_elementor_it_will_not_change() {
 		// Arrange
 		$admin = new Admin();
-		$post = $this->factory()->create_and_get_custom_post( [] );
+		$post  = $this->factory()->create_and_get_custom_post( [] );
```
Comment (matipojo) [inline tests/phpunit/elementor/core/admin/test-admin.php:52]: PHPCS

**Review by matipojo — CHANGES_REQUESTED** [review]

File: tests/phpunit/elementor/core/admin/test-admin.php (hunk 8 above)
Comment (Nevoss) [reply tests/phpunit/elementor/core/admin/test-admin.php:32]: Somehow it does not affect `$pagenow`, just checked it again.

**Review by matipojo — DISMISSED** [review]

**Review by kobizz — CHANGES_REQUESTED** [review]

**Review by matipojo — APPROVED** [review]

**Review by kobizz — APPROVED** [review]

**Pull request merged**